cpp/ScanChain_Zp.C

```diff
@@ -161,75 +177,73 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process) {
             if ( (year=="2017" || year=="2018") && !(nt.HLT_Mu50() || nt.HLT_OldMu100() || nt.HLT_TkMu100()) ) continue;
             h_cutflow->Fill(icutflow,weight*factor);
+	    h_cutflow->GetXaxis()->SetBinLabel(icutflow+1,"HLT");
             icutflow++;
 
             // Number of good primary vertices
             if ( nt.PV_npvsGood() < 1 ) continue;
             h_cutflow->Fill(icutflow,weight*factor);
+	    h_cutflow->GetXaxis()->SetBinLabel(icutflow+1,"At least 1 good PV");
             icutflow++;
 
             // Single muon selection loop
-            int nEvents_pt = 0;
-            int nEvents_iso = 0;
-            int nEvents_trk_glb = 0;
-            int nEvents_id = 0;
+            int nMu_id = 0;
+            int nMu_pt = 0;
+            int nMu_iso = 0;
             for ( unsigned int mu = 0; mu < nt.nMuon(); mu++ ){
-                bool mu_pt_pf = ( nt.Muon_pt().at(mu) > 53 );
-                bool mu_relIso = ( nt.Muon_tkRelIso().at(mu) < 0.1 );
-                bool mu_trk_and_global = ( nt.Muon_isGlobal().at(mu) && nt.Muon_isTracker().at(mu) );
-                bool mu_id = ( nt.Muon_highPtId().at(mu) == 2 );
-
-                if ( mu_pt_pf ){
-                    nEvents_pt++;
-                    if ( mu_relIso ){
-                        nEvents_iso++;	
-                        if ( mu_trk_and_global ){
-                            nEvents_trk_glb++;
-                            if ( mu_id ){
-                                nEvents_id++;
-                                cand_muons_pf.push_back(mu);
-				            }
-				        }
-				    }
-			    }
+	      bool mu_trk_and_global = ( nt.Muon_isGlobal().at(mu) && nt.Muon_isTracker().at(mu) );
+	      bool mu_id = ( nt.Muon_highPtId().at(mu) == 2 );
+	      bool mu_pt_pf = ( nt.Muon_pt().at(mu) > 53 );
+	      bool mu_relIso = ( nt.Muon_tkRelIso().at(mu) < 0.1 );
+	      
+	      if ( mu_trk_and_global && mu_id ){
+		nMu_id++;
+		if ( mu_pt_pf ){
+		  nMu_pt++;
+		  if ( mu_relIso ){
+		    nMu_iso++;
+		    cand_muons_pf.push_back(mu);
+		  }
+		}
+	      }
             }
+	    
+            // Defining booleans for cutflow
+            bool id_req = ( nMu_id > 1 );
+            bool pt_req = ( nMu_pt > 1 );
+            bool iso_req = ( nMu_iso > 1);
 
-            // Defining booleans for cutflow.....
-            bool pt_req = ( nEvents_pt > 1 );
-            bool iso_req = ( nEvents_iso > 1);
-            bool trk_glob_req = ( nEvents_trk_glb > 1 );
-            bool id_req = ( nEvents_id > 1 );
+            if ( !id_req ) continue;
+            h_cutflow->Fill(icutflow,weight*factor);
+	    h_cutflow->GetXaxis()->SetBinLabel(icutflow+1,"Muon ID (highPt)");
+            icutflow++;
 
             //Fill histograms before and after these requirements!
-            h_mu1_pt_pre->Fill(nt.Muon_pt().at(0),weight*factor);
-            h_mu2_pt_pre->Fill(nt.Muon_pt().at(1),weight*factor);
+            h_mu1_pt_sel1->Fill(nt.Muon_pt().at(0),weight*factor);
```

Comment:
Right. Since `cand_muons_pf` is filled only if all ID, pT and isolation requirements are fulfilled, I'll define two temporary vectors for muons satisfying only ID and ID & pT requirements respectively.

Reply:
Now that you mention it, I am wondering if there is the need for that, since this is only for a simple cross-check. Maybe we should just remove those plots? No strong preference.

Reply:
We can leave it there for the moment, but eventually I agree these can go.

Reply:
Done.